beeline-core/src/main/java/io/honeycomb/beeline/tracing/ids/W3CTraceIdProvider.java

```diff
@@ -0,0 +1,92 @@
+package io.honeycomb.beeline.tracing.ids;
+
+import java.util.Random;
+
+/**
+ * Creates Trace and Span IDs that conform to the W3C specification.
+ * This is inline with Beeline instrumentations in other programming languages.
+ *
+ * <h1>Thread-safety</h1>
+ * Instances of this class are thread-safe and can be shared.
+ */
+public class W3CTraceIdProvider implements TraceIdProvider {
+    private static final TraceIdProvider INSTANCE = new W3CTraceIdProvider();
+    private static final Random RAND = new Random();
```

Comment:
This is a pseudorandom number generator? Is there a better random number generator that we can use? Not a big deal if there isn't, ideally we would use the system RNG or a crypto RNG.

Reply:
This is the way a number of implementations (eg OpenTelemetry, beeline-go, etc) have gone too. I think we should go with this for now and I'll raise a ticket to review across all beelines.